golang/thrift/outbound.go

```diff
@@ -0,0 +1,323 @@
+package thrift
+
+import "bytes"
+import "errors"
+import "github.com/apache/thrift/lib/go/thrift"
+import tchannel "github.com/uber/tchannel/golang"
+import "golang.org/x/net/context"
+import "time"
+
+// NewTChannelOutboundProtocol creates a TChannelOutboundProtocol
+func NewTChannelOutboundProtocol(remoteHostPort, remoteServiceName, remoteProcessorName string) (*TChannelOutboundProtocol, error) {
+	tchannel, err := tchannel.NewChannel("0.0.0.0:0", nil)
```

Comment:
Yes. TChannelOutboundProtocol is not thread-safe, but can be re-used across multiple calls as long as these calls are from the same goroutine.

This does create a problem for the context stuff though. We may have to require a new TChannelOutboundProtocol for each call to get the context thing working.